tools/gulp/tasks/examples.ts

```diff
@@ -0,0 +1,234 @@
+import {task} from 'gulp';
+import {sync as glob} from 'glob';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as ts from 'typescript';
+import {buildConfig} from 'material2-build-tools';
+const {packagesDir} = buildConfig;
+
+interface ExampleMetadata {
+  component: string;
+  sourcePath: string;
+  id: string;
+  title: string;
+  additionalComponents: string[];
+  additionalFiles: string[];
+  selectorName: string[];
+}
+
+interface ParsedMetadata {
+  primary: boolean;
+  component: string;
+  title: string;
+  templateUrl: string;
+}
+
+interface ParsedMetadataResults {
+  primaryComponent: ParsedMetadata;
+  secondaryComponents: ParsedMetadata[];
+}
+
+/** Path to find the examples */
+const examplesPath = path.join(packagesDir, 'material-examples');
+
+/** Output path of the module that is being created */
+const outputModuleFilename = path.join(examplesPath, 'example-module.ts');
+
+/** Build ecmascript module import statements */
+function buildImportsTemplate(metadata: ExampleMetadata): string {
+  const components = metadata.additionalComponents.concat(metadata.component);
+
+  // Create a relative path to the source file of the current example.
+  // The relative path will be used inside of a TypeScript import statement.
+  const relativeSrcPath = path
+    .relative('./src/material-examples', metadata.sourcePath)
+    .replace(/\\/g, '')
+    .replace('.ts', '');
+
+  return `import {${components.join(',')}} from './${relativeSrcPath}';
+`;
+}
+
+/**
+ * Builds the examples metadata including title, component, etc.
+ */
+function buildExamplesTemplate(metadata: ExampleMetadata): string {
+  // if no additional files or selectors were provided,
+  // return undefined since we don't care about if these were not found
```

Comment:
Maybe change this comment to `null` now as well (since it's no longer `undefined` now). Maybe something like

```ts
// If there are no additional files or selectors specified, just set those variables
// to an empty array.
const additionalFiles = metadata.additionalFiles.length ? 
   JSON.stringify(metadata.additionalFiles) : '[]';
```

Not sure if an `empty` array would make sense here. It would allow us to drop more `null` checks in the docs app. But `null` is also fine.